db/migrate/20170629151616_add_unique_index_to_repos.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToRepos < ActiveRecord::Migration[5.1]
+  def change
+    add_index :repos, [:name, :user_name], :unique => true
```

Comment:
Great idea, honestly didn't know you could have a constraint without an index.

Looks like you can alter and add

	
A possible solution is to simply use DROP IF EXISTS before creating the new constraint.


    ALTER TABLE foo ADD CONSTRAINT bar ...;

Can use a unique constraint https://www.postgresql.org/docs/9.4/static/ddl-constraints.html